static/app/views/starfish/views/spans/useModuleSort.ts

```diff
@@ -0,0 +1,44 @@
+import {fromSorts} from 'sentry/utils/discover/eventView';
+import type {Sort} from 'sentry/utils/discover/fields';
+import {useLocation} from 'sentry/utils/useLocation';
+import {SpanMetricsFields, StarfishFunctions} from 'sentry/views/starfish/types';
+import {QueryParameterNames} from 'sentry/views/starfish/views/queryParameters';
+
+type Query = {
+  [QueryParameterNames.SORT]: string;
+};
+
+const SORTABLE_FIELDS = [
+  `avg(${SpanMetricsFields.SPAN_SELF_TIME})`,
+  `${StarfishFunctions.HTTP_ERROR_COUNT}()`,
+  `${StarfishFunctions.SPM}()`,
+  `${StarfishFunctions.TIME_SPENT_PERCENTAGE}()`,
+  `${StarfishFunctions.TIME_SPENT_PERCENTAGE}(local)`,
+] as const;
+
+export type ValidSort = Sort & {
+  field: (typeof SORTABLE_FIELDS)[number];
+};
+
+export const mySort: ValidSort = {
+  kind: 'asc',
+  field: 'spm()',
+};
+
+export function useModuleSort() {
+  const location = useLocation<Query>();
+
+  return (
+    fromSorts(location.query[QueryParameterNames.SORT]).filter(isAValidSort)[0] ??
+    DEFAULT_SORT
+  ); // We only allow one sort on this table in this view
```

Comment:
should this comment be moved into docstring for the hook?

Reply:
Done 👌🏻